crates/core_arch/src/mips/msa.rs

```diff
@@ -9178,1770 +9228,2189 @@ unsafe fn __msa_xori_b(a: u8x16, imm8: i32) -> u8x16 {
 
 #[cfg(test)]
 mod tests {
+    use crate::core_arch::mips::msa::*;
+    use core_arch::simd::*;
     use std::f32;
     use std::f64;
-    use core_arch::mips::msa::*;
+    use std::mem;
     use stdsimd_test::simd_test;
 
     #[simd_test(enable = "msa")]
     unsafe fn test_msa_add_a_b() {
         #[rustfmt::skip]
-        let a = i8x16(
+        let a = i8x16::new(
```

Comment:
Are there intrinsics to initialize the proper types directly? (not that there is anything bad about the transmutes, but it might be worth it to just use those).

Reply:
No, MIPS gcc-clang doesn't provide intrinsics for vector initialization, so I didn't want to create any new intrinsics for that purpose.

Reply:
Ok, well no big deal then - users can use a `union` of a vector and an array for initialization.